multiraft/storage.go

```diff
@@ -233,10 +233,17 @@ func (w *writeTask) start(stopper *stop.Stopper) {
 				groupResp := &groupWriteResponse{raftpb.HardState{}, -1, -1, groupReq.entries}
 				response.groups[groupID] = groupResp
 				if !raft.IsEmptySnap(groupReq.snapshot) {
-					err := group.ApplySnapshot(groupReq.snapshot)
-					if err != nil {
-						panic(err) // TODO(bdarnell)
-					}
+					gID := groupID
+					s.sendEvent(&EventSnapshot{
+						GroupID:  gID,
+						Snapshot: groupReq.snapshot,
+						Callback: func() {
```

Comment:
If this used `s.callbackChan` (like we do for `EventMembershipChangeCommitted`), the lock would not be needed and it would be more consistent with the rest of this file.